### PR TITLE
GA Migration Python PreCommit Sharding Runners Portability Spark

### PR DESCRIPTION
**Pull request by MarcoRob**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated workflow [job-precommit-python-runners-portability-spark.yml](https://github.com/apache/beam/blob/f46d489795ff68bbe5ed3c39b70015c6ce6a064e/.github/workflows/job-precommit-python-runners-portability-spark.yml)

The migrated workflow was added to [CI.md](https://github.com/apache/beam/blob/f46d489795ff68bbe5ed3c39b70015c6ce6a064e/CI.md)

`DO NOT MERGE` until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-precommit-python-runners-portability-spark.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: ['v*']
+    paths: [ 'sdks/python/apache_beam/runners/portability/spark_runner.py',
+             'sdks/python/apache_beam/runners/portability/spark_runner_test.py',
+             'sdks/python/apache_beam/runners/portability/spark_uber_jar_job_server.py',
+             'sdks/python/apache_beam/runners/portability/spark_uber_jar_job_server_test.py' ]
```
Comment (damccorm) [inline .github/workflows/job-precommit-python-runners-portability-spark.yml:32]: We should run postcommits on all paths, please remove these

**Review by damccorm — COMMENTED** [review]
Looks good other than one small comment

File: .github/workflows/job-precommit-python-runners-portability-spark.yml (hunk 1 above)
Comment (MarcoRob) [reply .github/workflows/job-precommit-python-runners-portability-spark.yml:32]: @damccorm same context, should we remove it? (considering this is a pre-commit job)

Comment (MarcoRob) [reply .github/workflows/job-precommit-python-runners-portability-spark.yml:32]: Hi @damccorm is there any comment on this?

**Pull request closed without merging**